### PR TITLE
"escapeTags" is removed from the addons

### PR DESCRIPTION
**Pull request by annando**
They had been used mostly for content that isn't displayed at all or that only an admin is allowed to enter, like in the impressum. So it was safe to remove the call.

### CODE REVIEW

File: buffer/buffer.php [hunk 1]
```diff
@@ -74,8 +73,8 @@ function buffer_addon_admin(App $a, &$o)
 
 function buffer_addon_admin_post(App $a)
 {
-	$client_id     = (!empty($_POST['client_id'])     ? Strings::escapeTags(trim($_POST['client_id']))     : '');
-	$client_secret = (!empty($_POST['client_secret']) ? Strings::escapeTags(trim($_POST['client_secret'])) : '');
+	$client_id     = (!empty($_POST['client_id'])     ? trim($_POST['client_id'])     : '');
+	$client_secret = (!empty($_POST['client_secret']) ? trim($_POST['client_secret']) : '');
```
Comment (nupplaphil) [inline buffer/buffer.php]:
```suggestion
	$client_id     = trim($_POST['client_id'] ?? '');
	$client_secret = trim($_POST['client_secret'] ?? '');
```

File: geocoordinates/geocoordinates.php [hunk 2]
```diff
@@ -91,9 +90,9 @@ function geocoordinates_addon_admin(&$a, &$o)
 
 function geocoordinates_addon_admin_post(&$a)
 {
-	$api_key  = (!empty($_POST['api_key']) ? Strings::escapeTags(trim($_POST['api_key']))   : '');
+	$api_key  = (!empty($_POST['api_key']) ? trim($_POST['api_key']) : '');
```
Comment (nupplaphil) [inline geocoordinates/geocoordinates.php]:
```suggestion
	$api_key  = trim($_POST['api_key'] ?? '');
```

File: geocoordinates/geocoordinates.php [hunk 3]
```diff
@@ -91,9 +90,9 @@ function geocoordinates_addon_admin(&$a, &$o)
...
 	DI::config()->set('geocoordinates', 'api_key', $api_key);
 
-	$language  = (!empty($_POST['language']) ? Strings::escapeTags(trim($_POST['language']))   : '');
+	$language  = (!empty($_POST['language']) ? trim($_POST['language']) : '');
```
Comment (nupplaphil) [inline geocoordinates/geocoordinates.php]:
```suggestion
	$language  = trim($_POST['language'] ?? '');
```

File: gravatar/gravatar.php [hunk 4]
```diff
@@ -108,8 +108,8 @@ function gravatar_addon_admin (&$a, &$o) {
 function gravatar_addon_admin_post (&$a) {
 	BaseModule::checkFormSecurityToken('gravatarsave');
 
-	$default_avatar = (!empty($_POST['avatar']) ? Strings::escapeTags(trim($_POST['avatar'])) : 'identicon');
-	$rating = (!empty($_POST['rating']) ? Strings::escapeTags(trim($_POST['rating'])) : 'g');
+	$default_avatar = (!empty($_POST['avatar']) ? trim($_POST['avatar']) : 'identicon');
+	$rating = (!empty($_POST['rating']) ? trim($_POST['rating']) : 'g');
```
Comment (nupplaphil) [inline gravatar/gravatar.php]:
```suggestion
	$default_avatar = trim($_POST['avatar'] ?? 'identicon');
	$rating = trim($_POST['rating'] ?? 'g');
```

File: impressum/impressum.php [hunk 5]
```diff
@@ -79,11 +78,11 @@ function impressum_show($a,&$b) {
 }
 
 function impressum_addon_admin_post (&$a) {
-    $owner = (!empty($_POST['owner']) ? Strings::escapeTags(trim($_POST['owner'])) : '');
-    $ownerprofile = (!empty($_POST['ownerprofile']) ? Strings::escapeTags(trim($_POST['ownerprofile'])) : '');
+    $owner = (!empty($_POST['owner']) ? trim($_POST['owner']) : '');
+    $ownerprofile = (!empty($_POST['ownerprofile']) ? trim($_POST['ownerprofile']) : '');
     $postal = (!empty($_POST['postal']) ? (trim($_POST['postal'])) : '');
     $notes = (!empty($_POST['notes']) ? (trim($_POST['notes'])) : '');
```
Comment (nupplaphil) [inline impressum/impressum.php]:
```suggestion
    $owner = trim($_POST['owner'] ?? '');
    $ownerprofile = trim($_POST['ownerprofile'] ?? '');
    $postal = trim($_POST['postal'] ?? '');
    $notes = trim($_POST['notes'] ?? '');
```

File: impressum/impressum.php [hunk 6]
```diff
@@ -79,11 +78,11 @@ function impressum_show($a,&$b) {
...
-    $email = (!empty($_POST['email']) ? Strings::escapeTags(trim($_POST['email'])) : '');
+    $email = (!empty($_POST['email']) ? trim($_POST['email']) : '');
     $footer_text = (!empty($_POST['footer_text']) ? (trim($_POST['footer_text'])) : '');
```
Comment (nupplaphil) [inline impressum/impressum.php]:
```suggestion
    $email = trim($_POST['email'] ?? '');
    $footer_text = trim($_POST['footer_text'] ?? '');
```

File: libravatar/libravatar.php [hunk 7]
```diff
@@ -90,6 +89,6 @@ function libravatar_addon_admin(&$a, &$o)
  */
 function libravatar_addon_admin_post(&$a)
 {
-	$default_avatar = (!empty($_POST['avatar']) ? Strings::escapeTags(trim($_POST['avatar'])) : 'identicon');
+	$default_avatar = (!empty($_POST['avatar']) ? trim($_POST['avatar']) : 'identicon');
```
Comment (nupplaphil) [inline libravatar/libravatar.php]:
```suggestion
	$default_avatar = trim($_POST['avatar'] ?? 'identicon');
```

File: newmemberwidget/newmemberwidget.php [hunk 8]
```diff
@@ -49,7 +48,7 @@ function newmemberwidget_network_mod_init ($a, $b)
 function newmemberwidget_addon_admin_post(&$a)
 {
 	$ft = (!empty($_POST['freetext']) ? trim($_POST['freetext']) : "");
-	$lsn = (!empty($_POST['localsupportname']) ? Strings::escapeTags(trim($_POST['localsupportname'])) : "");
+	$lsn = (!empty($_POST['localsupportname']) ? trim($_POST['localsupportname']) : "");
```
Comment (nupplaphil) [inline newmemberwidget/newmemberwidget.php]:
```suggestion
	$lsn = trim($_POST['localsupportname'] ?? '');
```

File: piwik/piwik.php [hunk 9]
```diff
@@ -102,7 +101,7 @@ function piwik_addon_admin (&$a, &$o) {
 	]);
 }
 function piwik_addon_admin_post (&$a) {
-	$url = (!empty($_POST['baseurl']) ? Strings::escapeTags(trim($_POST['baseurl'])) : '');
+	$url = (!empty($_POST['baseurl']) ? trim($_POST['baseurl']) : '');
 	$id = (!empty($_POST['siteid']) ? trim($_POST['siteid']) : '');
 	$optout = (!empty($_POST['optout']) ? trim($_POST['optout']) : '');
 	$async = (!empty($_POST['async']) ? trim($_POST['async']) : '');
```
Comment (nupplaphil) [inline piwik/piwik.php]:
```suggestion
	$url = trim($_POST['baseurl'] ?? '');
	$id = trim($_POST['siteid'] ?? '');
	$optout = trim($_POST['optout'] ?? '');
	$async = trim($_POST['async'] ?? '');
```

File: public_server/public_server.php [hunk 10]
```diff
@@ -126,12 +125,12 @@ function public_server_login($a, $b)
...
-	$nologin = (!empty($_POST['nologin']) ? Strings::escapeTags(trim($_POST['nologin'])) : '');
-	$flagusers = (!empty($_POST['flagusers']) ? Strings::escapeTags(trim($_POST['flagusers'])) : '');
-	$flagposts = (!empty($_POST['flagposts']) ? Strings::escapeTags(trim($_POST['flagposts'])) : '');
-	$flagpostsexpire = (!empty($_POST['flagpostsexpire']) ? Strings::escapeTags(trim($_POST['flagpostsexpire'])) : '');
+	$expiredays = (!empty($_POST['expiredays']) ? trim($_POST['expiredays']) : '');
+	$expireposts = (!empty($_POST['expireposts']) ? trim($_POST['expireposts']) : '');
+	$nologin = (!empty($_POST['nologin']) ? trim($_POST['nologin']) : '');
+	$flagusers = (!empty($_POST['flagusers']) ? trim($_POST['flagusers']) : '');
+	$flagposts = (!empty($_POST['flagposts']) ? trim($_POST['flagposts']) : '');
+	$flagpostsexpire = (!empty($_POST['flagpostsexpire']) ? trim($_POST['flagpostsexpire']) : '');
```
Comment (nupplaphil) [inline public_server/public_server.php]:
```suggestion
	$expiredays = trim($_POST['expiredays'] ?? '');
	$expireposts = trim($_POST['expireposts'] ?? '');
	$nologin = trim($_POST['nologin'] ?? '');
	$flagusers = trim($_POST['flagusers'] ?? '');
	$flagposts = trim($_POST['flagposts'] ?? '');
	$flagpostsexpire = trim($_POST['flagpostsexpire'] ?? '');
```

File: tumblr/tumblr.php [hunk 11]
```diff
@@ -76,8 +75,8 @@ function tumblr_addon_admin(App $a, &$o)
 
 function tumblr_addon_admin_post(App $a)
 {
-	$consumer_key    =       (!empty($_POST['consumer_key'])      ? Strings::escapeTags(trim($_POST['consumer_key']))   : '');
-	$consumer_secret =       (!empty($_POST['consumer_secret'])   ? Strings::escapeTags(trim($_POST['consumer_secret'])): '');
+	$consumer_key    = (!empty($_POST['consumer_key'])    ? trim($_POST['consumer_key'])   : '');
+	$consumer_secret = (!empty($_POST['consumer_secret']) ? trim($_POST['consumer_secret']): '');
```
Comment (nupplaphil) [inline tumblr/tumblr.php]:
```suggestion
	$consumer_key    = trim($_POST['consumer_key'] ?? : '');
	$consumer_secret = trim($_POST['consumer_secret'] ?? '');
```

File: twitter/twitter.php [hunk 12]
```diff
@@ -791,8 +791,8 @@ function (array $attributes, array $author_contact, $content, $is_quote_share) {
 
 function twitter_addon_admin_post(App $a)
 {
-	$consumerkey    = !empty($_POST['consumerkey'])    ? Strings::escapeTags(trim($_POST['consumerkey']))    : '';
-	$consumersecret = !empty($_POST['consumersecret']) ? Strings::escapeTags(trim($_POST['consumersecret'])) : '';
+	$consumerkey    = !empty($_POST['consumerkey'])    ? trim($_POST['consumerkey'])    : '';
+	$consumersecret = !empty($_POST['consumersecret']) ? trim($_POST['consumersecret']) : '';
```
Comment (nupplaphil) [inline twitter/twitter.php]:
```suggestion
	$consumerkey    = trim($_POST['consumerkey'] ?? : '');
	$consumersecret = trim($_POST['consumersecret'] ?? '');
```

File: webrtc/webrtc.php [hunk 13]
```diff
@@ -27,7 +26,7 @@ function webrtc_addon_admin (&$a, &$o) {
 	]);
 }
 function webrtc_addon_admin_post (&$a) {
-        $url = (!empty($_POST['webrtcurl']) ? Strings::escapeTags(trim($_POST['webrtcurl'])) : '');
+        $url = (!empty($_POST['webrtcurl']) ? trim($_POST['webrtcurl']) : '');
```
Comment (nupplaphil) [inline webrtc/webrtc.php]:
```suggestion
        $url = trim($_POST['webrtcurl'] ?? '');
```

**Review by nupplaphil — CHANGES_REQUESTED** [review]
Some PHP7 simplifications :)

Plz double-check e.g. if there are no missing brackets

**Comment by AlfredSK** [comment]
Error 500:

```
index [CRITICAL]: Uncaught Exception ParseError: "syntax error, unexpected ':'" at /var/www/html/libranet/addon/twitter/twitter.php line 794 {"exception":"ParseError: syntax error, unexpected ':' in /var/www/html/libranet/addon/twitter/twitter.php:794\nStack trace:\n#0 /var/www/html/libranet/src/Core/Hook.php(193): Friendica\\Core\\Hook::callSingle()\n#1 /var/www/html/libranet/src/App.php(375): Friendica\\Core\\Hook::callAll()\n#2 /var/www/html/libranet/src/App.php(346): Friendica\\App->load()\n#3 [internal function]: Friendica\\App->__construct()\n#4 /var/www/html/libranet/vendor/level-2/dice/Dice.php(132): ReflectionMethod->invokeArgs()\n#5 /var/www/html/libranet/vendor/level-2/dice/Dice.php(96): Dice\\Dice->Dice\\{closure}()\n#6 /var/www/html/libranet/src/DI.php(62): Dice\\Dice->create()\n#7 /var/www/html/libranet/index.php(39): Friendica\\DI::app()\n#8 {main}"} - {"file":null,"line":null,"function":null,"uid":"0cb1b7","process_id":3029970}
```

File: twitter/twitter.php [hunk 14]
```diff
@@ -791,8 +791,8 @@ function (array $attributes, array $author_contact, $content, $is_quote_share) {
 
 function twitter_addon_admin_post(App $a)
 {
-	$consumerkey    = !empty($_POST['consumerkey'])    ? Strings::escapeTags(trim($_POST['consumerkey']))    : '';
-	$consumersecret = !empty($_POST['consumersecret']) ? Strings::escapeTags(trim($_POST['consumersecret'])) : '';
+	$consumerkey    = trim($_POST['consumerkey'] ?? : '');
```
Comment (AlfredSK) [inline twitter/twitter.php:794]: The `:` produces an error 500 and WSOD.

Changing it to...
`$consumerkey    = trim($_POST['consumerkey'] ?? '');`
...solves the issue.

Comment (AlfredSK) [reply twitter/twitter.php:794]: Welp. This is only fixing the admin panel. Other pages still don't work.

Comment (AlfredSK) [reply twitter/twitter.php:794]: Tumblr has the same issue...

```
worker [CRITICAL]: Uncaught Exception ParseError: "syntax error, unexpected ':'" at /var/www/html/libranet/addon/tumblr/tumblr.php line 78 {"exception":"ParseError: syntax error, unexpected ':' in /var/www/html/libranet/addon/tumblr/tumblr.php:78\nStack trace:\n#0 /var/www/html/libranet/src/Core/Hook.php(193): Friendica\\Core\\Hook::callSingle()\n#1 /var/www/html/libranet/src/Model/Item.php(963): Friendica\\Core\\Hook::callAll()\n#2 /var/www/html/libranet/src/Model/Post/Delayed.php(230): Friendica\\Model\\Item::insert()\n#3 /var/www/html/libranet/src/Worker/DelayedPublish.php(42): Friendica\\Model\\Post\\Delayed::publish()\n#4 [internal function]: Friendica\\Worker\\DelayedPublish::execute()\n#5 /var/www/html/libranet/src/Core/Worker.php(497): call_user_func_array()\n#6 /var/www/html/libranet/src/Core/Worker.php(387): Friendica\\Core\\Worker::execFunction()\n#7 /var/www/html/libranet/src/Core/Worker.php(112): Friendica\\Core\\Worker::execute()\n#8 /var/www/html/libranet/bin/worker.php(86): Friendica\\Core\\Worker::processQueue()\n#9 {main}"} - {"file":null,"line":null,"function":null,"uid":"06f7d6","process_id":3076641}
```

Comment (AlfredSK) [reply twitter/twitter.php:794]: Fixing the unexpected `:` in both files solved it.

**Pull request merged**